todo_app/app.py

```diff
@@ -9,42 +9,58 @@
 import requests
 from todo_app.data.user import User
 from todo_app.data.roleChecker import ReadWriteNeeded
+from loggly.handlers import HTTPSHandler
+from logging import Formatter
 
 
 def create_app():
     app = Flask(__name__)
     app.config.from_object(Config())
     mongoDB = MongoDB()
     to_do_list_local = to_do_list(mongoDB)
+    app.logger.setLevel(app.config['LOG_LEVEL'])
+
+    if app.config['LOGGLY_TOKEN'] is not None:
+        handler = HTTPSHandler(f'https://logs-01.loggly.com/inputs/{app.config["LOGGLY_TOKEN"]}/tag/todo-app')
+        handler.setFormatter(Formatter("[%(asctime)s] %(levelname)s in %(module)s: %(message)s"))
+        app.logger.addHandler(handler)
 
     login_manager = LoginManager()
 
+
     @login_manager.unauthorized_handler
     def unauthenticated():
         url = f'https://github.com/login/oauth/authorize?client_id={os.getenv("GIT_HUB_CLIENT_ID")}'
         return redirect(url)
 
     @login_manager.user_loader
     def load_user(user_id):
-        return User(user_id)
+        user = User(user_id)
+        app.logger.info("User is " + user.role)
+        return user
+
 
     login_manager.init_app(app)
 
     @app.route('/login/callback')
     def login_callback():
-        url = 'https://github.com/login/oauth/access_token'
-        code = request.args['code']
-        params = {'client_id':os.getenv('GIT_HUB_CLIENT_ID'), 'client_secret':os.getenv('GIT_HUB_SECRET'), 'code':code }
-        header = {'Accept': 'application/json'}
-        oauth_token = requests.post(url, params=params, headers=header).json()['access_token']
+        try:
+            url = 'https://github.com/login/oauth/access_token'
+            code = request.args['code']
+            params = {'client_id':os.getenv('GIT_HUB_CLIENT_ID'), 'client_secret':os.getenv('GIT_HUB_SECRET'), 'code':code }
+            header = {'Accept': 'application/json'}
+            oauth_token = requests.post(url, params=params, headers=header).json()['access_token']
+
+            api_url = 'https://api.github.com/user'
+            api_header = {'Accept':'application/vnd.github+json', 'Authorization': f'Bearer {oauth_token}'}
+            user_id = requests.get(api_url, headers=api_header).json()['id']
+            user = User(user_id)
+            login_user(user)
+            return redirect('/')
+        except:
+            app.logger.critical('Unable to log in')
```

Comment:
I'd be inclined to make this an `error` rather than a `critical` - if it's happening for everyone then it might be critical, but it seems likely to me that once a day my app might block a user login and I wouldn't be worried about it, whereas critical typically means the system is crashing and needs immediate attention